packages/fetchai/connections/p2p_noise/aea/api.go

```diff
@@ -0,0 +1,395 @@
+package aea
+
+import (
+  "fmt"
+  "os"
+  "encoding/binary"
+  proto "github.com/golang/protobuf/proto"
+  "math"
+  "math/rand"
+  "time"
+  "errors"
+  "syscall"
+  "strconv"
+  "net"
+  "strings"
+)
+
+/*
+  
+  AeaApi type
+
+*/
+
+type AeaApi struct {
+  msgin_path  string
+  msgout_path string
+  id          string
+  entry_uris  []string
+  host        net.IP
+  port        uint16
+  msgin       *os.File
+  msgout      *os.File
+  out_queue   chan *Envelope
+  closing     bool
+  sandbox     bool
+}
+
+func (aea AeaApi) PrivateKey() string {
+  return aea.id
+}
+
+func (aea AeaApi) Uri() (net.IP, uint16) {
+  return aea.host, aea.port
+}
+
+func (aea AeaApi) EntryUris() []string {
+  return aea.entry_uris
+}
+
+func (aea AeaApi) Put(envelope *Envelope) error {
+  return write_envelope(aea.msgout, envelope)
+}
+
+func (aea *AeaApi) Get() *Envelope {
+  return <- aea.out_queue
+}
+
+func (aea *AeaApi) Queue() <-chan *Envelope {
+  return aea.out_queue
+}
+
+func (aea *AeaApi) Stop() {
+  aea.closing = true
+  aea.stop()
+  close(aea.out_queue)
+}
+
+func (aea *AeaApi) Init() error {
+  if aea.sandbox {
+    return nil
+  }
+
+  // get config
+  aea.msgin_path  = os.Getenv("AEA_TO_NOISE")
+  aea.msgout_path = os.Getenv("NOISE_TO_AEA")
+  aea.id  = os.Getenv("ID")
+  entry_uris := os.Getenv("ENTRY_URI")
+  uri := os.Getenv("URI")
+
+  if aea.msgin_path == "" || aea.msgout_path == "" || aea.id == "" || uri == "" {
+    fmt.Println("[aea-api  ][error] couldn't get configuration")
```

Comment:
instead of just printing to stdout, would be useful to employ some Go logging library like https://golang.org/pkg/log/. That would let the developer have finer control over the messages printed by the package, at _configuration level_.

Reply:
I agree. 
I was planning to use the default one (the one you point it out), but *due to the lack of warn/debug/trace levels* I kept printing to stdout until I find a better logger.